src/server/publicApi.js

```diff
@@ -8,198 +8,218 @@ import { API_PORT, INTERNAL_API_PORT } from '../utils/constants';
 import { getTypeString, escapeMarkdownText } from '../utils/utils';
 
 const runPublicApi = (gameServer) => {
-
-    const app = new Koa();
-    const router = new Router();
-
-    router.get('/players/:id', async ctx => {
-    const gameID = ctx.params.id;
-    const r = await request
-        .get(`http://localhost:${INTERNAL_API_PORT}/games/${ElevationOfPrivilege.name}/${gameID}`);
+  const app = new Koa();
```

Comment:
Am I correct, that everything (except for the last line) is only changing the indentation?

Reply:
No, I've made a few changes but mainly just changing the names of functions/variables, What I'll do is write a quick summary of what I've done.

Maybe I should've held off reformatting it until after you'd checked it out

Reply:
- Requests to the internal API now use `matchID` instead of `gameID`
- `gameServer.db.set()` has now been replaced by `gameServer.db.setState()`, `setMetadata()` or `setModel()` respectively
- `gameServer.db.get()` has been replaced with `gameServer.db.fetch()`
  - The syntax of this is to provide the matchID and the fields you want returned
  - e.g. `fetch(matchID, {state:true, metadata:true})` will return an object with the properties state and metadata
    - it returns something like this `{state: /* Current value of G */, metadata: /* current metadata */}`

Reply:
For using the database connector, there's no documentation but [this](https://github.com/boardgameio/boardgame.io/blob/ce8ef4a16bcc420b05c5e0751b41f168352bce7d/src/server/db/base.ts#L49-L111) is the interface it implements although it doesn't really describe the behaviour.

The best way to see how the connector should behave is to look at the FlatFile implementation in node_modules/boardgame.io/cjs/server.js

```javascript
/**
 * FlatFile data storage.
 */
class FlatFile extends Async {
    constructor({ dir, logging, ttl }) {
        super();
        this.games = require('node-persist');
        this.dir = dir;
        this.logging = logging || false;
        this.ttl = ttl || false;
        this.fileQueues = {};
    }
    async chainRequest(key, request) {
        if (!(key in this.fileQueues))
            this.fileQueues[key] = Promise.resolve();
        this.fileQueues[key] = this.fileQueues[key].then(request, request);
        return this.fileQueues[key];
    }
    async getItem(key) {
        return this.chainRequest(key, () => this.games.getItem(key));
    }
    async setItem(key, value) {
        return this.chainRequest(key, () => this.games.setItem(key, value));
    }
    async removeItem(key) {
        return this.chainRequest(key, () => this.games.removeItem(key));
    }
    async connect() {
        await this.games.init({
            dir: this.dir,
            logging: this.logging,
            ttl: this.ttl,
        });
        return;
    }
    /**
     * Create new match.
     *
     * @param matchID
     * @param opts
     * @override
     */
    async createMatch(matchID, opts) {
        // Store initial state separately for easy retrieval later.
        const key = InitialStateKey(matchID);
        await this.setItem(key, opts.initialState);
        await this.setState(matchID, opts.initialState);
        await this.setMetadata(matchID, opts.metadata);
    }
    async fetch(matchID, opts) {
        const result = {};
        if (opts.state) {
            result.state = (await this.getItem(matchID));
        }
        if (opts.metadata) {
            const key = MetadataKey(matchID);
            result.metadata = (await this.getItem(key));
        }
        if (opts.log) {
            const key = LogKey(matchID);
            result.log = (await this.getItem(key));
        }
        if (opts.initialState) {
            const key = InitialStateKey(matchID);
            result.initialState = (await this.getItem(key));
        }
        return result;
    }
    async clear() {
        return this.games.clear();
    }
    async setState(id, state, deltalog) {
        if (deltalog && deltalog.length > 0) {
            const key = LogKey(id);
            const log = (await this.getItem(key)) || [];
            await this.setItem(key, log.concat(deltalog));
        }
        return await this.setItem(id, state);
    }
    async setMetadata(id, metadata) {
        const key = MetadataKey(id);
        return await this.setItem(key, metadata);
    }
    async wipe(id) {
        const keys = await this.games.keys();
        if (!keys.includes(id))
            return;
        await this.removeItem(id);
        await this.removeItem(InitialStateKey(id));
        await this.removeItem(LogKey(id));
        await this.removeItem(MetadataKey(id));
    }
    /**
     * List matches IDs.
     *
     * @param opts
     * @override
     */
    async listMatches(opts) {
        const keys = await this.games.keys();
        const suffix = ':metadata';
        const arr = await Promise.all(keys.map(async (k) => {
            if (!k.endsWith(suffix)) {
                return false;
            }
            const matchID = k.slice(0, k.length - suffix.length);
            if (!opts) {
                return matchID;
            }
            const game = await this.fetch(matchID, {
                state: true,
                metadata: true,
            });
            if (opts.gameName && opts.gameName !== game.metadata.gameName) {
                return false;
            }
            if (opts.where !== undefined) {
                if (typeof opts.where.isGameover !== 'undefined') {
                    const isGameover = typeof game.metadata.gameover !== 'undefined';
                    if (isGameover !== opts.where.isGameover) {
                        return false;
                    }
                }
                if (typeof opts.where.updatedBefore !== 'undefined' &&
                    game.metadata.updatedAt >= opts.where.updatedBefore) {
                    return false;
                }
                if (typeof opts.where.updatedAfter !== 'undefined' &&
                    game.metadata.updatedAt <= opts.where.updatedAfter) {
                    return false;
                }
            }
            return matchID;
        }));
        return arr.filter((r) => typeof r === 'string');
    }
}
function InitialStateKey(matchID) {
    return `${matchID}:initial`;
}
function MetadataKey(matchID) {
    return `${matchID}:metadata`;
}
function LogKey(matchID) {
    return `${matchID}:log`;
}
```